src/solvers/smt2/smt2_conv.cpp

```diff
@@ -2274,6 +2275,17 @@ void smt2_convt::convert_typecast(const typecast_exprt &expr)
     else
       UNEXPECTEDCASE("Unknown typecast "+src_type.id_string()+" -> float");
   }
+  else if(dest_type.id()==ID_integer)
+  {
+    if(src_type.id()==ID_bool)
+    {
+      out << "(ite ";
+      convert_expr(src);
+      out <<" 1 0 )";
```

Comment:
Nit picking: `out << " 1 0)";` to put the space in the right places.

Reply:
Done